clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecords.java

```diff
@@ -41,12 +41,12 @@ public ConsumerRecords(Map<TopicPartition, List<ConsumerRecord<K, V>>> records)
      * 
      * @param partition The partition to get records for
      */
-    public Iterable<ConsumerRecord<K, V>> records(TopicPartition partition) {
+    public List<ConsumerRecord<K, V>> records(TopicPartition partition) {
```

Comment:
It's convenient to be able to look at the head or tail member, which you can't do with an Iterable without iterating through all the members. Unless we think there's some chance that we'd change the collection type in the future, it seemed more convenient to return the specific type.